tools/plugingenerator/plugin_generator.py

```diff
@@ -0,0 +1,229 @@
+import json
+import os
+import argparse
+import shutil
+from mako.template import Template
+
+parser = argparse.ArgumentParser(
+    prog="Plugin generator",
+    description="Support script for plugin development",
+    epilog="Text at the bottom of help",
+)
+
+parser.add_argument("--scopy_path", help="The absolute path to the Scopy repository")
+
+args = parser.parse_args()
+path = ""
+if args.scopy_path:
+    path = args.scopy_path
+else:
+    path = os.path.dirname(os.path.dirname(os.getcwd()))
+
+configFile = open("config.json", "r")
+generatorOptions = json.load(configFile)
+configFile.close()
+
+mode = 0o775
+
+pluginDirName = generatorOptions["plugin"]["dir_name"]
+pluginName = generatorOptions["plugin"]["plugin_name"]
+pluginClassName = generatorOptions["plugin"]["class_name"]
+pluginExportMacro = "SCOPY_" + pluginName.upper() + "_EXPORT"
+
+pluginsPath = os.path.join(path, "plugins")
+if not os.path.exists(pluginsPath):
+    print("Couldn't find " + pluginsPath + " path!")
+    exit(1)
+
+#################################################### Plugin dir #############################################
+newPluginPath = os.path.join(pluginsPath, pluginDirName)
+if not os.path.exists(newPluginPath):
+    os.mkdir(newPluginPath, mode)
+    print("plugin directory created")
```

Comment:
Include full path for all directories and files created to better show what exactly is being done.